src/main/java/net/blay09/mods/cookingforblockheads/block/BlockCounterCorner.java

```diff
@@ -88,12 +88,18 @@ public boolean recolourBlock(World world, int x, int y, int z, ForgeDirection si
     @Override
     public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer player, int side, float hitX, float hitY, float hitZ) {
         ItemStack heldItem = player.getHeldItem();
+        TileCounterCorner corner = (TileCounterCorner) world.getTileEntity(x, y, z);
+
         if(heldItem != null && DyeUtils.isDye(heldItem)) {
             Optional<Integer> dyeColor = DyeUtils.colorFromStack(heldItem);
-            if (dyeColor.isPresent() && recolourBlock(world, x, y, z, ForgeDirection.UNKNOWN, dyeColor.get())) {
```

Comment:
Another approach would be to have `recolourBlock` return false if the color hasn't changed.